lib/room.rb

```diff
@@ -0,0 +1,46 @@
+require "date"
+
+class Room
+  attr_reader :id, :price, :bookings, :booked_on
+  ROOMS = 20
+
+  def initialize(id:, bookings: nil)
+    unless id.instance_of?(Integer) && id > 0 && id <= 20
+      raise ArgumentError, "ID must be a positive number, given #{id}..."
+    end
+
+    @id = id
+    @bookings ||= []
+  end
+
+  def self.hotel_rooms
+    rooms = []
+    ROOMS.times do |room_num|
+      rooms << Room.new(id: room_num + 1)
+    end
+    return rooms
+  end
+
+  def booked_on(check_in:, check_out:)
+    @bookings << (Date.parse(check_in)..Date.parse(check_out))
+  end
+
+  def room_available?(check_in:, check_out:)
+    check_in = Date.parse(check_in)
+    check_out = Date.parse(check_out)
+    @bookings.each do |booked|
```

Comment:
I get really confused by what `booked` means in this loop, I think because it's a verb and not a noun. I'd have gone with `booking`, or `old_booking` for clarity.